src/commands/database-get.ts

```diff
@@ -91,70 +100,45 @@ export default new Command("database:get <path>")
     );
 
     const urlObj = new url.URL(dbUrl);
-    Object.keys(query).forEach((key) => {
-      urlObj.searchParams.set(key, query[key]);
+    const client = new Client({
+      urlPrefix: urlObj.origin,
+      auth: true,
+    });
+    const res = await client.request<unknown, NodeJS.ReadableStream>({
+      method: "GET",
+      path: urlObj.pathname,
+      queryParams: query,
+      responseType: "stream",
+      resolveOnHTTPError: true,
     });
 
-    dbUrl = urlObj.href;
-
-    logger.debug("Query URL: ", dbUrl);
-    const reqOptions = {
-      url: dbUrl,
-    };
+    const fileOut = !!options.output;
+    const outStream = fileOut ? fs.createWriteStream(options.output) : process.stdout;
 
-    return api.addRequestHeaders(reqOptions).then((reqOptionsWithToken) => {
-      return new Promise((resolve, reject) => {
-        const fileOut = !!options.output;
-        const outStream = fileOut ? fs.createWriteStream(options.output) : process.stdout;
-        const writeOut = (s: Buffer | string, cb?: Function): void => {
-          if (outStream === process.stdout) {
-            outStream.write(s, cb);
-          } else if (outStream instanceof fs.WriteStream) {
-            outStream.write(s, (err) => {
-              if (cb) {
-                cb(err);
-              }
-            });
-          }
-        };
-        let erroring = false;
-        let errorResponse = "";
-        let response: any;
+    if (res.status >= 400) {
+      const r = await utils.streamToString(res.body);
```

Comment:
Can we move this to `src/responseToError.js`? See comment above.

Reply:
Yes, I think that's possible. But if I'm going to do that, responseToError is going to be converted to TS for my sanity. I can take this as a TODO to follow up on for sure.

Some of me is considering that I shouldn't though, if only because it's not necessarily obvious what we _should_ be doing with the streamed error content. In this case, we know that the 4xx response should be well-formed JSON, but other (more custom - maybe downloading emulators) may not return JSON. It can still _try_ I suppose, but that's my thought process - we definitely can always manually handle errors!

Reply:
Ack. See comment above.